### DISCUSSION TITLE
Does Portman provide support for conversion of models referenced by a specification

### DISCUSSION THREAD

**Discussion by PaulCluver** [apideck-libraries/portman#493] [Q&A] [answered]
Hi there, 

My specification document references models certain models via $ref. Does Portman provide the ability to generate these models?  

Kind regards,
Paul.

**Comment by thim81** [comment, marked as answer]
@PaulCluver 

Portman dereferences the OpenAPI documents before converting them, so if the references are accessible, they will get included in the conversion.

**Reply by PaulCluver** [reply]
Thanks for your prompt response. I am still a bit baffled as to why this is not working. I can only assume that this is a pathing issue. Perhaps the references are not accessible. I get the following as output when executing my portman cli command. 

`ResolverError: Error opening file "C:\dev\work\sahl.testing.api-management\tmp\converted\models\calculator-v2.yml"`

**Reply by thim81** [reply]
The error points indeed like an issue with the file locally not being available when resolving the $ref.

**Reply by cw112233** [reply]
I have met the same issue. I have root OAS referencing to other OAS under the same path. But when I tried to execute portman command, I noticed only the root OAS has been moved to `tmp\converted\` folder, thus also generating the resolverError
